praw/exceptions.py

```diff
@@ -1,38 +1,137 @@
+
+    @property
+    def error_message(self):
+        """Get the completed error message string."""
+        error_str = "{}: {!r}".format(self.error_type, self.message)
+        if self.field:
+            error_str += " on field {!r}".format(self.field)
+        return error_str
 
-    def __init__(self, error_type: str, message: str, field: Optional[str]):
-        """Initialize an instance of APIException.
+    def __init__(
+        self, error_type: str, message: str, field: Optional[str] = None
+    ):
+        """Instantiate an error item.
 
         :param error_type: The error type set on Reddit's end.
         :param message: The associated message for the error.
-        :param field: The input field associated with the error if available.
+        :param field: The input field associated with the error, if available.
         """
-        error_str = "{}: '{}'".format(error_type, message)
-        if field:
-            error_str += " on field '{}'".format(field)
-
-        super().__init__(error_str)
         self.error_type = error_type
         self.message = message
         self.field = field
 
+    def __eq__(self, other: Union[_RedditErrorItem, List[str]]):
+        """Check for equality."""
+        if isinstance(other, RedditErrorItem):
+            return (self.error_type, self.message, self.field) == (
+                other.error_type,
+                other.message,
+                other.field,
+            )
+        return super().__eq__(other)
+
+    def __repr__(self):
+        """Return repr(self)."""
+        return "{}(error_type={!r}, message={!r}, field={!r})".format(
+            self.__class__.__name__, self.error_type, self.message, self.field
+        )
+
+    def __str__(self):
+        """Get the message returned from str(self)."""
+        return self.error_message
+
+
+class RedditAPIException(PRAWException):
+    """Container for error messages from Reddit's API."""
+
+    @staticmethod
+    def parse_exception_list(
+        exceptions: List[Union[RedditErrorItem, List[str]]]
+    ):
+        """Covert an exception list into a :class:`.RedditErrorItem` list."""
+        return [
+            exception
+            if isinstance(exception, RedditErrorItem)
+            else RedditErrorItem(
+                exception[0],
+                exception[1],
+                exception[2] if bool(exception[2]) else "",
+            )
+            for exception in exceptions
+        ]
+
+    def __init__(self, items: List[Union[RedditErrorItem, List[str]]]):
+        """Initialize an instance of RedditAPIException.
+
+        :param items: Either a list of instances of :class:`.RedditErrorItem`
+            or a list containing lists of unformed errors.
+        """
+        self.items = self.parse_exception_list(items)
+        super().__init__(*self.items)
+
+
+class APIException(RedditAPIException):
+    """Old class preserved for alias purposes."""
+
+    @property
```

Comment:
Since all new exceptions will be of type `RedditAPIException`, these attributes won't be available there. To work, shouldn't they be defined on `RedditAPIException` instead?

Reply:
Oh right, I forgot new excepts won't have these attributes. These really should go to the base class instead.